### DISCUSSION TITLE
Statically linking C code.

### DISCUSSION THREAD

**Discussion by Joe23232** [GodotECS/godex#279] [General]
Hey guys I wanted to ask you in regards to Godot ECS, is it going to have a more Bevy-like approach when it comes to development?

Is it possible for me to code in C and have everything statically linked rather than creating a GDNative C code?

**Comment by AndreaCatania** [comment]
Right now Godex is only statically compiled, and even if the plan is to support dynamic compilation via GDExtension, we will always support static compilation.

**Reply by Joe23232** [reply]
@AndreaCatania 

That actually sounds great.

Is there a lot of boilerplate code when coding in Godex (using C) or is it easy to understand and use just like with Godot GDScript?

**Reply by AndreaCatania** [reply]
Godex is implemented using C++. If you want to use pure C, it's impossible without some additional work to allow that. However, if it's fine for you mix C and C++, you could already register systems and component using C++, but inside the systems you can just use C.
